Mathlib/Analysis/Normed/Ring/SmoothingSeminorm.lean

```diff
@@ -0,0 +1,625 @@
+      (fun n => rpow_nonneg (apply_nonneg _ _) _) (μ_bddAbove μ hμ1 hnu_le _ _)
+        fun n => rpow_nonneg (apply_nonneg _ _) _).isBoundedUnder_of_range hxy hε
+  obtain ⟨N, hN⟩ := hex
+  apply le_trans (ciInf_le (smoothingSeminormSeq_bddBelow μ _)
+    ⟨ψ N, lt_of_le_of_lt (_root_.zero_le (ψ 0)) (hψ_mono.lt_iff_lt.mpr N.pos)⟩)
+  apply le_trans _ hN.le
+  simp only [PNat.mk_coe, hnu, ← mul_rpow (apply_nonneg μ _) (apply_nonneg μ _)]
+  exact mu_property μ hn (ψ N)
+
+end IsNonarchimedean
+
+/-- If `μ 1 ≤ 1` and `μ` is nonarchimedean, then `smoothingFun` is a ring seminorm. -/
+def smoothingSeminorm (hμ1 : μ 1 ≤ 1) (hna : IsNonarchimedean μ) : RingSeminorm R where
+  toFun     := smoothingFun μ
+  map_zero' := by
+    apply tendsto_nhds_unique_of_eventuallyEq (tendsto_smoothingFun_of_map_one_le_one μ hμ1 0)
+      tendsto_const_nhds
+    simp only [EventuallyEq, eventually_atTop, ge_iff_le]
+    use 1
+    intro n hn
+    simp only [smoothingSeminormSeq]
+    rw [zero_pow (pos_iff_ne_zero.mp hn), map_zero, zero_rpow]
+    exact one_div_ne_zero (cast_ne_zero.mpr (one_le_iff_ne_zero.mp hn))
+  add_le' _ _ :=  (isNonarchimedean_smoothingFun μ hμ1 hna).add_le (smoothingFun_nonneg μ hμ1)
+  neg' n := by
+    simp only [smoothingFun]
+    congr
+    ext n
+    rw [neg_pow]
+    rcases neg_one_pow_eq_or R n with hpos | hneg
+    · rw [hpos, one_mul]
+    · rw [hneg, neg_one_mul, map_neg_eq_map μ]
+  mul_le' x y := by
+    apply le_of_tendsto_of_tendsto' (tendsto_smoothingFun_of_map_one_le_one μ hμ1 (x * y))
+      (Tendsto.mul (tendsto_smoothingFun_of_map_one_le_one μ hμ1 x)
+        (tendsto_smoothingFun_of_map_one_le_one μ hμ1 y))
+    intro n
+    have hn : 0 ≤ 1 / (n : ℝ) := by simp only [one_div, inv_nonneg, cast_nonneg]
+    simp only [smoothingSeminormSeq]
+    rw [← mul_rpow (apply_nonneg μ _) (apply_nonneg μ _), mul_pow]
+    exact rpow_le_rpow (apply_nonneg μ _) (map_mul_le_mul μ _ _) hn
+
+/-- If `μ 1 ≤ 1` and `μ` is nonarchimedean, then `smoothingSeminorm μ hμ1 hna 1 ≤ 1`. -/
+theorem smoothingSeminorm_map_one_le_one (hμ1 : μ 1 ≤ 1)
+    (hna : IsNonarchimedean μ) : smoothingSeminorm μ hμ1 hna 1 ≤ 1 :=
+  smoothingFun_one_le μ hμ1
+
+/-- If `μ 1 ≤ 1` and `μ` is nonarchimedean, then `smoothingFun μ` is
+  power-multiplicative. -/
+theorem isPowMul_smoothingFun (hμ1 : μ 1 ≤ 1) : IsPowMul (smoothingFun μ) := by
+  intro x m hm
+  have hlim : Tendsto (fun n => smoothingSeminormSeq μ x (m * n)) atTop
+      (𝓝 (smoothingFun μ x)) :=
+    Tendsto.comp (tendsto_smoothingFun_of_map_one_le_one μ hμ1 x) (tendsto_atTop_atTop_of_monotone
+      (fun n k hnk ↦ mul_le_mul_left' hnk m) (fun n ↦ ⟨n, le_mul_of_one_le_left' hm⟩))
+  apply tendsto_nhds_unique _ (Tendsto.pow hlim m)
+  have h_eq : ∀ n : ℕ, smoothingSeminormSeq μ x (m * n) ^ m = smoothingSeminormSeq μ (x ^ m) n := by
+    have hm' : (m : ℝ) ≠ 0 := cast_ne_zero.mpr (_root_.ne_of_gt (lt_of_lt_of_le zero_lt_one hm))
+    intro n
+    simp only [smoothingSeminormSeq]
```

Comment:
```suggestion
  have h_eq (n : ℕ) : smoothingSeminormSeq μ x (m * n) ^ m = smoothingSeminormSeq μ (x ^ m) n := by
    have hm' : (m : ℝ) ≠ 0 := cast_ne_zero.mpr (_root_.ne_of_gt (lt_of_lt_of_le zero_lt_one hm))
    simp only [smoothingSeminormSeq]
```

Reply:
Oh, simply because the first line seemed to exceed the 100 chars and removing the \forall sign helps with this. Not really serious...

Reply:
I think maybe my comment was not at the right place - I was asking about why to remove the `simp only` on line 336.

Reply:
I think I suggested to remove the `intro n` (I think it is not line 336 anymore, right?)